optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -121,66 +125,268 @@
      * @param otherClass never null
      * @param joiner never null
      * @param <B> the type of the second matched fact
-     * @return a stream that matches every combination of A and B for which the {@link BiJoiner} is true
+     * @return never null, a stream that matches every combination of A and B for which the {@link BiJoiner} is true
      */
     default <B> BiConstraintStream<A, B> join(Class<B> otherClass, BiJoiner<A, B> joiner) {
-        return join(getConstraintFactory().from(otherClass), joiner);
+        if (joiner instanceof FilteringBiJoiner) {
+            FilteringBiJoiner<A, B> filteringJoiner = (FilteringBiJoiner<A, B>) joiner;
+            return join(otherClass)
+                    .filter(filteringJoiner.getFilter());
+        } else {
+            return join(getConstraintFactory().from(otherClass), joiner);
+        }
     }
 
     /**
-     * As defined by {@link #join(Class, BiJoiner)}.
+     * As defined by {@link #join(Class, BiJoiner...)}.
```

Comment:
This puts the vararg method front-and-center. This has 2 issues:

1)The vararg method suffers from "Unchecked generics array creation for varargs parameter" JDK warnings, which we can only fix by making the method non-abstract (which is obviously impossible). Therefore, I see the method as a necessary evil: not something we want to put front-and-center.
2) This javadoc delegates to the vararg method javadoc, which in turn delegates to the single joiner method. Every extra click through looses users. Yet we do delegates due to reduce maintenance burden - and to make users that already know the base method, be able to "start from there". For varargs that latest approach doesn't work.
3) vararg's javadoc is NOT a subset of this javadocs. For example, that warning doesn't apply here at all.

Reply:
I recommend going back to delegating to the single joiner param, wdyt?